src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -144,16 +112,14 @@ class WorkspaceNewRoomPage extends React.Component {
                         onCloseButtonPress={() => Navigation.dismissModal()}
                     />
                     <ScrollView style={styles.flex1} contentContainerStyle={styles.p5}>
-                        <TextInputWithLabel
-                            label={this.props.translate('newRoomPage.roomName')}
-                            prefixCharacter="#"
-                            placeholder={this.props.translate('newRoomPage.social')}
-                            containerStyles={[styles.mb5]}
-                            onChangeText={roomName => this.setState({roomName: this.checkAndModifyRoomName(roomName)})}
-                            value={this.state.roomName.substr(1)}
-                            errorText={this.state.error}
-                            autoCapitalize="none"
-                        />
+                        <View style={styles.mb5}>
+                            <Text style={[styles.formLabel]}>{this.props.translate('newRoomPage.roomName')}</Text>
+                            <RoomNameInput
+                                onChangeText={(roomName) => { this.setState({roomName}); }}
+                                initialValue={this.state.roomName}
+                                policyID={this.state.policyID}
+                            />
+                        </View>
```

Comment:
Why are we un-componentizing this?

Reply:
I did it because the label is used differently between WorkspaceNewRoomPage and RoomSettingsPage.

In the RoomSettingsPage we want the "Save" button to be under the NewRoomLabel too, but the WorkspaceNewRoomPage doesn't need the "Save" button. Since one needed the save button and the other needed the label used this way. I decided to just keep exactly what was common between the two in the RoomNameInput component.

![image](https://user-images.githubusercontent.com/19364431/149024819-6a72ddf9-fb3a-4fb0-ab90-07fed1001bb3.png)
![image](https://user-images.githubusercontent.com/19364431/149024847-2d247984-8705-4f6a-adc1-7b873c1f756e.png)